test/plugin/test_out_google_cloud.rb

```diff
@@ -191,6 +191,30 @@ def test_parse_severity
     assert_equal('DEFAULT', test_obj.parse_severity('er'))
   end
 
+  def test_non_integer_timestamp
+    setup_gce_metadata_stubs
+    time = Time.now
+    [
+      { 'seconds' => nil, 'nanos' => nil },
+      { 'seconds' => nil, 'nanos' => time.tv_nsec },
+      { 'seconds' => nil, 'nanos' => time.tv_nsec },
+      { 'seconds' => 'seconds', 'nanos' => time.tv_nsec },
+      { 'seconds' => time.tv_sec, 'nanos' => 'nanos' },
+      { 'seconds' => time.tv_sec, 'nanos' => nil }
+    ].each_with_index do |input, index|
+      setup_logging_stubs do
+        d = create_driver
+        @logs_sent = []
+        d.emit('message' => log_entry(0), 'timestamp' => input)
+        d.run
+      end
+      verify_log_entries(1, COMPUTE_PARAMS) do |entry|
+        assert_equal input, entry['metadata']['timestamp'],
+                     "Index #{index} failed with entry: #{entry.inspect}"
```

Comment:
Ah, you're right. Then perhaps use the stringified input instead of the index in both places?